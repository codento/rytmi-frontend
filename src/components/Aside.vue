<template>
  <aside class="aside-menu" />
</template>

<script>
export default {
  name: 'c-aside'
}
</script>
