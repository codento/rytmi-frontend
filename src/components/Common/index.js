import ProjectRow from './ProjectRow.vue'
import SkillRow from './SkillRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import UtilizationChart from './UtilizationChart.vue'
import CollapsableItem from './CollapsableItem.vue'
import EducationCard from './EducationCard.vue'
import ProjectForm from './ProjectForm.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectRow,
  SkillRow,
  SkillExplanations,
  UtilizationChart,
  CollapsableItem,
  EducationCard,
  ProjectForm,
  ProjectProfileForm
}
