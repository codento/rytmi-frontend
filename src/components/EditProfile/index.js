import EditLinks from './EditLinks.vue'
import EditSkills from './EditSkills.vue'
import ProfileForm from './ProfileForm.vue'
import EditProjects from './EditProjects.vue'
import WorkHistoryEntryEdit from './WorkHistoryEntryEdit.vue'
import WorkHistoryEntryList from './WorkHistoryEntryList.vue'
import CvInfo from './CvInfo.vue'
import CvInfoEditIntroduction from './CvInfoEditIntroduction.vue'
import CvInfoEditEducation from './CvInfoEditEducation.vue'
import CvInfoEditEducationForm from './CvInfoEditEducationForm.vue'

export {
  EditLinks,
  EditSkills,
  ProfileForm,
  EditProjects,
  WorkHistoryEntryList,
  CvInfo,
  CvInfoEditIntroduction,
  CvInfoEditEducation,
  CvInfoEditEducationForm,
  WorkHistoryEntryEdit
}
