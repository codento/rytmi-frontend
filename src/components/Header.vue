<template>
  <header class="app-header navbar">
    <button class="navbar-toggler mobile-sidebar-toggler d-lg-none" type="button" @click="mobileSidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    <b-link class="navbar-brand" to="Home"></b-link>
    <button class="navbar-toggler sidebar-toggler d-md-down-none" type="button" @click="sidebarToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
     <b-navbar-nav  v-if="isAuthenticated">
      <b-nav-item to="/search">Search</b-nav-item>

      <b-nav-item v-if="!getProfileId" to="/createProfile">Create Profile</b-nav-item>
      <b-nav-item v-else to="/editProfile">Edit My Profile</b-nav-item>

    </b-navbar-nav>
    <b-navbar-nav class="ml-auto">
      <b-nav-item id="loginBtn"  v-if="!isAuthenticated" v-on:click="login" >Sign in</b-nav-item>
      <b-nav-item id="logoutBtn" v-if="isAuthenticated" v-on:click="logout" >Sign out</b-nav-item>
      <b-nav-item id="logoutBtn"  v-on:click="checkStatus" >Check status</b-nav-item>
    </b-navbar-nav>
    <button class="navbar-toggler aside-menu-toggler d-md-down-none" type="button" @click="asideToggle">
      <span class="navbar-toggler-icon"></span>
    </button>
    
  </header>
</template>
<script>
import {mapGetters, mapActions} from 'vuex'
import {loadAuthClient, handleLogin, handleLogout} from '../utils/auth'

export default {
  name: 'c-header',
  computed: {
    ...mapGetters(['isAuthenticated', 'getProfileId', 'getToken'])
  },
  methods: {
    sidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-hidden')
    },
    sidebarMinimize (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-minimized')
    },
    mobileSidebarToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('sidebar-mobile-show')
    },
    asideToggle (e) {
      e.preventDefault()
      document.body.classList.toggle('aside-menu-hidden')
    },
    checkStatus () {
      this.$toasted.global.rytmi_success({
        message: 'Authenticated: ' + this.isAuthenticated
      })
    },
    logout () {
      handleLogout().then(() => {
        this.logoutAuth()
        this.$router.push(this.$route.query.redirect || '/home')
      })
    },
    login () {
      handleLogin().then((response) => {
        this.requestAuth(response.Zi.id_token)
      }).catch((error) => {
        console.log(error)
        this.$toasted.global.rytmi_error({
          message: 'Login failed'
        })
      })
    },
    ...mapActions(['requestAuth', 'logoutAuth', 'createUser'])
  },
  created () {
    /* eslint-disable */
    loadAuthClient()
  }
}
</script>
