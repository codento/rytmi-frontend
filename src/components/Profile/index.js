import ProjectRow from './ProjectRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import SkillRow from './SkillRow.vue'
import UserProfileCard from './UserProfileCard.vue'

export {
  ProjectRow,
  SkillExplanations,
  SkillRow,
  UserProfileCard
}
