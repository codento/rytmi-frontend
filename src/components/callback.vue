<template>
  
</template>

<script>
import {mapActions} from 'vuex'

export default {
  created () {
    this.$nextTick(() => {
      this.fetchProfiles()
      this.fetchSkills()
      this.fetchProfileSkills()
      this.$router.push(this.$route.query.redirect || '/')
    })
  },
  methods: {
    ...mapActions([
      'fetchProfiles',
      'fetchSkills',
      'fetchProfileSkills'
    ])
  }
}
</script>