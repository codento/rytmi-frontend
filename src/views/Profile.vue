<template>
  <b-container class="animated fadeIn profile">
    <loading v-if="!profile"/>
    <b-row v-else>
      <b-col class="col-md-6 col-sm-12 col-12">
        <UserProfileCard :profile="profile"/>
      </b-col>
      <b-col class="col-md-6 col-sm-12 col-12">
        <b-card id="proficiency">
          <h5
            slot="header"
            class="mb-0">
            Proficiency
            <span
              style="float:right; cursor:pointer"
              @mouseenter="showExplanations(true)"
              @mouseout="showExplanations(false)">
              &#9432;
            </span>
          </h5>
          <b-row v-if="showInfo">
            <skillExplanations
              :know-desc="knowDesc"
              :want-desc="wantDesc" />
          </b-row>
          <b-row v-else>
            <b-col class="col mb-1">
              <SkillRow
                v-for="skill in skillsByProfileId(profile.id)"
                :name="skillName(skill.skillId)"
                :knows="skill.knows"
                :wants="skill.wantsTo"
                :desc="skill.description"
                :key="skill.id"/>
            </b-col>
          </b-row>
        </b-card>
        <b-card header="Projects">
          <loading v-if="!profileProjects"/>
          <ProjectRow
            v-for="profileProject in profileProjects"
            v-else
            :profile-project="profileProject"
            :key="profileProject.id"
          />
        </b-card>
        <b-card header="Utilization">
          <loading v-if="!profileProjects"/>
          <UtilizationChart
            v-else
            :projects="profileProjects" />
        </b-card>
      </b-col>
    </b-row>
  </b-container>
</template>

<script>
import { mapGetters } from 'vuex'
import store from '../store'
import proficiencyDesc from '../assets/proficiencyDesc'
import {
  ProjectRow,
  SkillExplanations,
  SkillRow,
  UserProfileCard,
  UtilizationChart
} from '../components/Profile'

export default {
  name: 'Profile',
  components: {
    SkillRow,
    ProjectRow,
    UserProfileCard,
    SkillExplanations,
    UtilizationChart
  },
  data () {
    return {
      showInfo: false
    }
  },
  computed: {
    ...mapGetters([
      'profileById',
      'skillById',
      'skillsByProfileId',
      'profileProjectsByProfileId'
    ]),
    profile () {
      return this.profileById(this.$route.params.id)
    },
    knowDesc () {
      return proficiencyDesc.knows
    },
    wantDesc () {
      return proficiencyDesc.wants
    },
    profileProjects () {
      return this.profileProjectsByProfileId(this.profile.id)
    }
  },
  watch: {
    profile: function (val, oldVal) {
      if (val) {
        document.title = `Rytmi - ${val.firstName} ${val.lastName}`
      }
    }
  },
  methods: {
    getSkills () {
      return this.skillsByProfileId(this.profile.userId)
    },
    sortSkills (param) {
      return this.$lodash.orderBy(this.skills, ['wantsTo', 'name'], ['desc', 'asc']) // TODO Figure out how lodash should be handled in tests
    },
    skillName (skillId) {
      return this.skillById(skillId).name
    },
    showExplanations (show) {
      this.showInfo = show
    }
  },
  beforeRouteEnter (to, from, next) {
    store.dispatch('fetchPPsOfProfile', to.params.id)
    next()
  }
}
</script>
