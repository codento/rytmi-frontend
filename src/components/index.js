import Aside from './appStructures/Aside.vue'
import Breadcrumb from './appStructures/Breadcrumb.vue'
import Callout from './helpers/Callout.vue'
import Footer from './appStructures/Footer.vue'
import Header from './appStructures/Header.vue'
import Sidebar from './appStructures/Sidebar/Sidebar.vue'

export {
  Aside,
  Breadcrumb,
  Callout,
  Footer,
  Header,
  Sidebar
}
