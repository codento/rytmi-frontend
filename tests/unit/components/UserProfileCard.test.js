import {
  shallowMount,
  createLocalVue
} from '@vue/test-utils'
import Vuex from 'vuex'
import { UserProfileCard } from '../../../src/components/Profile'

describe('UserProfileCard.test.js', () => {
  let localVue = createLocalVue()
  localVue.use(Vuex)
  let store, getters

  beforeEach(() => {
    getters = {
      profileId: () => (arg) => arg,
      profileById: () => (arg) => arg,
      profiles: () => (arg) => arg,
      skillsById: () => (arg) => arg,
      skillsByProfileId: () => (arg) => arg,
      profileProjectsByProfileId: () => (arg) => arg
    }
    store = new Vuex.Store({ getters })
  })
  it('Template is correct', () => {
    let wrapper = shallowMount(UserProfileCard, {
      store,
      localVue,
      propsData: {
        profile: {
          name: 'test',
          email: 'mail',
          phone: '1231123',
          mail: 'mail@mail.comi',
          accounts: null,
          photoPath: 'shouldNotWork'
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
