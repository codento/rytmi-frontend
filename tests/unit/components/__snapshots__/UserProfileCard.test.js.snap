// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileCard.test.js Template is correct 1`] = `
<div
  class="profile-details"
>
  <div
    style="text-align: center;"
  >
    <img
      alt="profile photo"
      src="shouldNotWork"
    />
  </div>
   
  <div
    style="text-align: center; color: rgb(134, 159, 172);"
  >
    <span
      class="profile-name"
    >
       undefined undefined
    </span>
    <br />
     
    <span
      class="profile-title"
    >
      
    </span>
     
  </div>
   
  <div>
    <div
      class="profileCardDetails profile-card-detail-row"
    >
      
      mail
    
    </div>
     
    <div
      class="profileCardDetails profile-card-detail-row"
    >
      
      1231123
    
    </div>
     
    <br />
     
    <div
      class="profileCardDetails profile-card-detail-row"
    >
      
      
    
    </div>
     
    <!---->
  </div>
</div>
`;
